### ISSUE TITLE
[Feature Request] :  Add radius to any corner of a bar chart

### ISSUE THREAD

**Issue by abiodunooni** [Abhinandan-Kushwaha/react-native-gifted-charts#354]
Thank you for the great package.

**Describe the bug**
In some use case, it is necessary for me to add a border radius to the top-left and top-right corners of a bar chart without adding any radius to the bottom corners.

There is currently no way to achieve this (that I'm aware of). The current options that exist is wither to round the top, or apply a border radius to all the corners. What I'm looking for is more like the below:

<img width="138" alt="Screenshot 2023-08-05 at 12 37 55 AM" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/59513570/2faa2a95-3b12-47d5-9b91-1a820e531160">

**Comment by Abhinandan-Kushwaha** [comment]
Hi @abiodunooni 👋
Thanks for requesting the feature. Currently, we don't support `borderTopLeftRadius` and `borderTopRightRadius`. I will add it in the next update.

Till then I can suggest you a workaround. Just add prop `cappedBars` and then set the border-radius using the `capRadius` prop. If your capRadius is bigger than 6, you also need to set the `capThickness` equal to the capRadius.

Here's an example-
```js
<BarChart
  data={data}
  cappedBars
  capRadius={8}
  capThickness={8} />
```

**Comment by giltheodor** [comment]
Is this feature available for stacked bars? For example can I add a border radius to the top-left and top-right corners of a bar chart without adding any radius to the bottom corners.
![Frame 7458](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/111180498/24e5f4e4-6a0d-489a-8d29-1ac565bec98f)

**Comment by Abhinandan-Kushwaha** [comment]
Hi @giltheodor 
For stacked bar charts, you don't need to send `barBorderTopLeftRadius` and `brBorderTopRightRadius` as props.
Instead you should add the properties `borderTopLeftRadius` and `borderTopRightRadius` inside the objects of `stacks` array of `stackData` prop.

**Comment by giltheodor** [comment]
This solution is working fine but if the top bar in the stackbar has height:0 then the 
![Uploading Simulator Screenshot - iPhone 15 - 2023-11-28 at 12.10.53.png…]()
style is not visible.

**Comment by Abhinandan-Kushwaha** [comment]
Noted @giltheodor 
Thanks for requesting this feature. It will be rolled out in the next release by tomorrow.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @giltheodor 👋

Added these props to apply border-radius to Stacked Bar charts- 
<ul>
   <li> stackBorderRadius </li>
   <li> stackBorderTopLeftRadius </li>
   <li> stackBorderTopRightRadius </li>
   <li> stackBorderBottomLeftRadius </li>
   <li> stackBorderBottomRightRadius </li>
</ul>

These will apply border-radius to the top and/or bottom item of each stack.  While props like `barBorderRadius`, `barBorderTopLeftRadius`... will apply border-radius to each item of each stack.

Additionally, you can pass `borderRadius`, `borderTopLeftRadius`... to any particular section or stack of the `stackData` array, so the border will be applied only to that section.

This feature is available from version `1.3.24` onwards.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @giltheodor 
Thanks for pointing out the issue. `stackBorderRadius` and `stackBorderTopLeftRadius` etc were not working properly if the value of the top item in a stack was very small. It is now fixed in version `1.3.25` 🎉
